routers/routes/routes.go

```diff
@@ -878,7 +878,11 @@ func RegisterRoutes(m *macaron.Macaron) {
 			m.Get("/:period", repo.ActivityAuthors)
 		}, context.RepoRef(), repo.MustBeNotEmpty, context.RequireRepoReaderOr(models.UnitTypeCode))
 
-		m.Get("/archive/*", repo.MustBeNotEmpty, reqRepoCodeReader, repo.Download)
+		m.Group("/archive", func() {
+			m.Post("/status/*", repo.DownloadStatus)
```

Comment:
Can this generate a name conflict? (e.g. a user named `status`)

Reply:
Shouldn't- this is a little further down in the chain: `/:username/:reponame/archive/status/*`